packages/font/src/google/fetch-resource.ts

```diff
@@ -0,0 +1,56 @@
+import http from 'node:http'
+import https from 'node:https'
+import { getProxyAgent } from './get-proxy-agent'
+
+/**
+ * Makes a simple GET request and returns the entire response as a Buffer.
+ * - Throws if the response status is not 200.
+ * - Applies a 3000 ms timeout when `isDev` is `true`.
+ */
+export function fetchResource(
+  url: string,
+  isDev: boolean,
+  errorMessage?: string
+): Promise<Buffer> {
+  return new Promise((resolve, reject) => {
+    const { protocol } = new URL(url)
+    const client = protocol === 'https:' ? https : http
+    const timeout = isDev ? 3000 : undefined
+
+    const req = client.request(
+      url,
+      {
+        agent: getProxyAgent(),
+        headers: {
+          'User-Agent':
```

Comment:
maybe worth preserving the previous comment about why a user agent is specified here